### PR TITLE
Slack notifications service for Milestones behind flag

### PR DESCRIPTION
**Pull request by aileen**
refs https://www.notion.so/ghost/Marketing-Milestone-email-campaigns-1d2c9dee3cfa4029863edb16092ad5c4?pvs=4

- Added a `slack-notifications` repository which handles sending Slack messages to a URL as defined in our Ghost(Pro) config (also includes a global switch to disable the feature if needed) and listens to `MilestoneCreatedEvents`.
- Added a `slack-notification` service which listens to the events on boot.
- In order to have access to further information such as the reason why a Milestone email hasn't been sent, or the current ARR or Member value as comparison to the achieved milestone, I added a `meta` object to the `MilestoneCreatedEvent` which then gets accessible by the event subscriber. This avoid doing further requests to the DB as we need to have this information in relation to the event occurred.

### CODE REVIEW

**Comment by aileen** [comment]
Going to add an e2e server test as well

File: ghost/core/core/server/services/slack-notifications/service.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const DomainEvents = require('@tryghost/domain-events');
+const config = require('../../../shared/config');
+const labs = require('../../../shared/labs');
+const logging = require('@tryghost/logging');
+
+class SlackNotificationsServiceWrapper {
+    /** @type {import('@tryghost/slack-notifications/lib/SlackNotificationsService')} */
+    #api;
+
+    static create({siteUrl, isEnabled, webhookUrl}) {
```
Comment (aileen) [inline ghost/core/core/server/services/slack-notifications/service.js:10]: Will add JSDoc here. Forgot!

Comment (aileen) [reply ghost/core/core/server/services/slack-notifications/service.js:19]: @allouis not sure if `@returns {object} SlackNotificationsService` this is correct 😬

File: ghost/core/test/e2e-server/services/milestones.test.js [hunk 2]
```diff
@@ -151,9 +151,6 @@ describe('Milestones Service', function () {
 
     beforeEach(async function () {
         sinon.createSandbox();
-        // TODO: stub out stripe mode
```
Comment (aileen) [inline ghost/core/test/e2e-server/services/milestones.test.js:154]: Left over todo, forgot to remove when I added the stubs for it

File: ghost/core/core/server/services/slack-notifications/service.js (hunk 1 above)
Comment (allouis) [reply ghost/core/core/server/services/slack-notifications/service.js:19]: Just `@returns {SlackNotificationsService}` ☺️

**Pull request merged**